controllers/util/version.go

```diff
@@ -0,0 +1,125 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	semver "github.com/blang/semver/v4"
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	// TODO: update the fixed in kernel versions
+	// supported kernel version corresponding to the OCP version
+	CephxKeyRotaionKernelSupportMatrix = map[string]string{
+		"4.14": "5.14.0-570.22.1.el9_6.x86_64",
```

Comment:
The last parts of these versions are not interesting, `.el9_6` and `.x86_64` can be dropped from the version comparing check. RHEL kernels always have those last two parts.

`570.22.1` is not guaranteed to be only numeric. Hotfix kernels often refer to a particular bug by having `.bz123456` or similar in it.

Reply:
I will drop this `el9_6.x86_64`. We should be able to compare even in the case of a hotfix and all.

Reply:
not any more, looks good now.